src/main/java/space/obminyashka/items_exchange/service/impl/UserServiceImpl.java

```diff
@@ -192,7 +191,7 @@ private boolean isDurationMoreThanNumberOfDaysToKeepDeletedUser(User user) {
 
     @Override
     public void makeAccountActiveAgain(User user) {
-        user.setStatus(ACTIVE);
+        roleService.updateUserRole(user.getUsername());
         userRepository.saveAndFlush(user);
```

Comment:
Here are a couple of comments on the whole method:
1. You don't need to save and flush the user when you update him in **updateUserRole** method.
2. You have to pass fewer privileges to the method. If you need only a username here pass only it, no need to pass the whole user.